internal/controller/driver_controller.go

```diff
@@ -17,46 +17,1246 @@ limitations under the License.
+								Image:           r.images["provisioner"],
+								Args: append(
+									slices.Clone(leaderElectionArgs),
+									utils.LogLevelContainerArg(logLevel),
+									utils.CsiAddressContainerArg,
+									utils.TimeoutContainerArg(grpcTimeout),
+									utils.RetryIntervalStartContainerArg,
+									utils.DefaultFsTypeContainerArg,
+									utils.PreventVolumeModeConversionContainerArg,
+									utils.HonorPVReclaimPolicyContainerArg,
+									utils.If(r.isRdbDriver(), utils.DefaultFsTypeContainerArg, ""),
+									utils.If(r.isRdbDriver(), utils.TopologyContainerArg, ""),
+									utils.If(!r.isNfsDriver(), utils.ExtraCreateMetadataContainerArg, ""),
+								),
+								VolumeMounts: []corev1.VolumeMount{
+									utils.SocketDirVolumeMount,
+								},
+								Resources: ptr.Deref(
+									pluginSpec.Resources.Provisioner,
+									corev1.ResourceRequirements{},
+								),
+							},
+							// Resizer Sidecar Container
+							{
+								Name:            "csi-resizer",
+								ImagePullPolicy: imagePullPolicy,
+								Image:           r.images["resizer"],
+								Args: append(
+									slices.Clone(leaderElectionArgs),
+									utils.LogLevelContainerArg(logLevel),
+									utils.CsiAddressContainerArg,
+									utils.TimeoutContainerArg(r.driver.Spec.GRpcTimeout),
+									utils.HandleVolumeInuseErrorContainerArg,
+									utils.RecoverVolumeExpansionFailureContainerArg,
+								),
+								VolumeMounts: []corev1.VolumeMount{
+									utils.SocketDirVolumeMount,
+								},
+								Resources: ptr.Deref(
+									pluginSpec.Resources.Resizer,
+									corev1.ResourceRequirements{},
+								),
+							},
+							// Attacher Sidecar Container
+							{
+								Name:            "csi-attacher",
+								ImagePullPolicy: imagePullPolicy,
+								Image:           r.images["attacher"],
+								Args: append(
+									slices.Clone(leaderElectionArgs),
+									utils.LogLevelContainerArg(logLevel),
+									utils.CsiAddressContainerArg,
+									utils.TimeoutContainerArg(grpcTimeout),
+									utils.If(r.isRdbDriver(), utils.DefaultFsTypeContainerArg, ""),
+								),
+								VolumeMounts: []corev1.VolumeMount{
+									utils.SocketDirVolumeMount,
+								},
+								Resources: ptr.Deref(
+									pluginSpec.Resources.Attacher,
+									corev1.ResourceRequirements{},
+								),
+							},
+							// Snapshotter Sidecar Container
+							{
+								Name:            "csi-snapshotter",
+								ImagePullPolicy: imagePullPolicy,
+								Image:           r.images["snapshotter"],
+								Args: append(
+									slices.Clone(leaderElectionArgs),
+									utils.LogLevelContainerArg(logLevel),
+									utils.CsiAddressContainerArg,
+									utils.TimeoutContainerArg(grpcTimeout),
+									utils.If(r.isNfsDriver(), utils.ExtraCreateMetadataContainerArg, ""),
+									utils.If(
+										r.driverType != NfsDriverType,
+										utils.EnableVolumeGroupSnapshotsContainerArg,
```

Comment:
This will not work for older images because this flag needs to be set only if volumegroupsnapshot CRDs are present and if the sidecar image supports this flag.

Reply:
What happens if the image does not support it? will the arg be ignored or would the command fail?

Reply:
container will not start it will crash

Reply:
That is unfortunate, it means that we need to support a capability table for each Ceph SDI release! 
And we will need to have it in sync for each change in Ceph CSI sidecars. That will be a never-ending chase. 

With that said we cannot have the operator deploying containers that indefinitely crush. 
We need a plan here, and maybe involvement of the core Ceph CSI team itself

Reply:
As discussed offline, we need to address this one with container argument checks, opening a tracker for it.